lib/jobs/dell-racadm-tool-job.js

```diff
@@ -41,32 +41,32 @@ function racadmSetBIOSJobFactory(
      * @param {String} taskId
      * @constructor
      */
-    function RacadmSetBiosJob(options, context, taskId) {
-        RacadmSetBiosJob.super_.call(this, logger, options, context, taskId);
+    function RacadmToolJob(options, context, taskId) {
+        RacadmToolJob.super_.call(this, logger, options, context, taskId);
 
         this.nodeId = this.context.target;
-        this.cifsInfo = {user: options.username, password: options.password ,
-            filePath: options.filePath};
-
+        this.action = options.action;
+        this.cifsInfo = {user: options.serverUsername, password: options.serverPassword ,
+            filePath: options.serverFilePath};
     }
-    util.inherits(RacadmSetBiosJob, BaseJob);
+    util.inherits(RacadmToolJob, BaseJob);
 
     /**
-     * @memberOf RacadmSetBiosJob
+     * @memberOf RacadmToolJob
      */
-    RacadmSetBiosJob.prototype._run = function() {
+    RacadmToolJob.prototype._run = function() {
         var self = this;
         waterline.nodes.findByIdentifier(self.nodeId)
             .then(function (node) {
-                assert.ok(node, 'No node for dell racadm set bios');
+                assert.ok(node, 'No node for dell racadm tool');
                 var obmSetting = _.find(node.obmSettings, { service: 'ipmi-obm-service' });
-                assert.ok(obmSetting, 'No ipmi obmSetting for dell racadm set bios');
+                assert.ok(obmSetting, 'No ipmi obmSetting for dell racadm tool');
                 return obmSetting.config;
             })
             .then(self.revealSecrets)
             .then(self.lookupHost)
             .then(function(obmSetting) {
-                return racadmTool.setBiosConfig(obmSetting.host, obmSetting.user,
+                return racadmTool[self.action](obmSetting.host, obmSetting.user,
```

Comment:
Might be good to assert that `racadmTool[self.action]` is a function first, unless I'm missing this somewhere.